crates/sncast/src/starknet_commands/script/init.rs

```diff
@@ -0,0 +1,159 @@
+    );
+
+    Ok(scripts_dir.join(script_name))
+}
+
+fn init_scarb_project(script_name: &str, script_root_dir: &Utf8PathBuf) -> Result<()> {
+    ScarbCommand::new()
+        .args([
+            "new",
+            "--name",
+            &script_name,
+            "--no-vcs",
+            "--quiet",
+            script_root_dir.as_str(),
+        ])
+        .run()
+        .context("Failed to init Scarb project")?;
+
+    Ok(())
+}
+
+fn add_dependencies(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    add_sncast_std_dependency(script_root_dir)
+        .context("Failed to add sncast_std dependency to Scarb.toml")?;
+    add_starknet_dependency(script_root_dir)
+        .context("Failed to add starknet dependency to Scarb.toml")?;
+
+    Ok(())
+}
+
+fn add_sncast_std_dependency(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    let cast_version = format!("v{}", env!("CARGO_PKG_VERSION"));
+
+    ScarbCommand::new()
+        .current_dir(script_root_dir)
+        .args([
+            "--offline",
+            "add",
+            "sncast_std",
+            "--git",
+            "https://github.com/foundry-rs/starknet-foundry.git",
+            "--tag",
+            &cast_version,
+        ])
+        .run()?;
+
+    Ok(())
+}
+
+fn add_starknet_dependency(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    let scarb_manifest_path = script_root_dir.join("Scarb.toml");
+    let cairo_version =
+        get_cairo_version(&scarb_manifest_path).context("Failed to get cairo version")?;
+    let starknet_dependency = format!("starknet@>={cairo_version}");
+
+    ScarbCommand::new()
+        .current_dir(script_root_dir)
+        .args(["--offline", "add", &starknet_dependency])
+        .run()?;
+
+    Ok(())
+}
+
+fn modify_files_in_src_dir(script_name: &str, script_root_dir: &Utf8PathBuf) -> Result<()> {
+    create_script_main_file(script_name, script_root_dir)
+        .context(format!("Failed to create {script_name}.cairo file"))?;
+    overwrite_lib_file(script_name, script_root_dir).context("Failed to overwrite lib.cairo file")
+}
+
+fn create_script_main_file(script_name: &str, script_root_dir: &Utf8PathBuf) -> Result<()> {
+    let script_main_file_name = format!("{script_name}.cairo");
+    let script_main_file_path = script_root_dir.join("src").join(script_main_file_name);
+
+    fs::write(
+        script_main_file_path,
+        indoc! {r#"
+            use sncast_std::{call, CallResult};
+
+            // The example below uses a contract deployed to the Goerli testnet
+            fn main() {
+                let contract_address = 0x7ad10abd2cc24c2e066a2fee1e435cd5fa60a37f9268bfbaf2e98ce5ca3c436;
+                let call_result = call(contract_address.try_into().unwrap(), 'get_greeting', array![]);
+                let call_result = *call_result.data[0];
+                let mut greeting: ByteArray = "";
+                greeting.append_word(call_result, 16);
+                println!("{greeting:?}");
```

Comment:
As I mentioned before this not an ideal way to print a shortstring. In the [ByteArrayImpl](https://github.com/starkware-libs/cairo/blob/main/corelib/src/byte_array.cairo#L49-L50) we can read that `append_word` should be a private function.